app/ui-react/syndesis/cypress/integration/create-connection.spec.js

```diff
@@ -0,0 +1,146 @@
+        cy.get('[data-testid=connection-card-delete-action]').click();
+      }
+    );
+
+    cy.get('#deleteConfirmationDialogContent').should('be.visible');
+    cy.get('.modal-footer')
+      .contains('Delete')
+      .click();
+
+    cy.get('.toast-pf.alert-success').should('be.visible');
+  }
+
+  /**
+   * Happy Path
+   *
+   * 1. User navigates to Create Connection page from the Dashboard.
+   * 2. User is able to see a list of Connectors.
+   * 3. User selects a Connector, and is redirected to Step 2.
+   * 4. User is able to populate configuration fields in Step 2, with validation.
+   * 5. User selects Next button to move to Step 3.
+   * 6. User is able to add a name and description in Step 3, with validation.
+   * 7. User is able to click Save to create Connection.
+   * 8. User is redirected to Connections list page and sees newly created Connection.
+   */
+
+  it('loads Step 1: Select connector page', () => {
+    cy.visit('/');
+
+    cy.get('[data-testid=dashboard-create-connection-button]').click();
+
+    cy.location('pathname').should(
+      'contain',
+      '/connections/create/connection-basics'
+    );
+
+    cy.get('.active > a > .wizard-pf-step-number').should('have.text', '1');
+  });
+
+  it('loads Step 2: Configure connection page with DB connector', () => {
+    /**
+     * DB connector
+     */
+    cy.get('[data-testid=connection-card-database-card]')
+      .should('exist')
+      .click();
+
+    cy.location('pathname').should(
+      'contain',
+      '/connections/create/sql/configure-fields'
+    );
+
+    cy.get('.active > a > .wizard-pf-step-number').should('have.text', '2');
+
+    cy.get('[data-testid=connection-creator-layout-next-button]').should(
+      'be.disabled'
+    );
+
+    cy.get('[data-testid=url]').focus();
+
+    cy.get('[data-testid=user]').focus();
+
+    cy.get('.form-group.has-error').should('be.visible');
+
+    cy.get('[data-testid=url]')
+      .clear()
+      .type(Cypress.env('connectorDbUrl'))
+      .should('have.value', Cypress.env('connectorDbUrl'));
+
+    cy.get('[data-testid=user]')
+      .clear()
+      .type(Cypress.env('connectorDbUser'))
+      .should('have.value', Cypress.env('connectorDbUser'));
+
+    cy.get('[data-testid=password]')
+      .clear()
+      .type(Cypress.env('connectorDbPassword'))
+      .should('have.value', Cypress.env('connectorDbPassword'));
+
+    cy.get('[data-testid=schema]')
+      .clear()
+      .type(Cypress.env('connectorDbSchema'))
+      .should('have.value', Cypress.env('connectorDbSchema'));
+
+    cy.get('[data-testid=connection-creator-layout-next-button]')
+      .should('not.be.disabled')
+      .click();
+  });
+
+  it('ensures the page renders properly', () => {
+    /**
+     * Sidebar should be collapsed
+     */
+    cy.get('.pf-c-page__sidebar-body').should('be.hidden');
+
+    /**
+     * Test Cancellation modal
+     */
+    cy.get('[data-testid=connection-creator-layout-cancel-button]')
```

Comment:
Should this test maybe be in it's own scenario specific to testing the cancel dialog?

Reply:
This is where it gets tricky, that's probably what the unit tests should do. I am trying to keep E2E tests page-specific as opposed to functionality-specific, but the lines kind of start to blur when there is overlapping functionality. I can maybe separate it out into a shared spec for those cases.

Reply:
Then again, it's also good to know that the the cancellation modal works from the perspective of the user in each page. Even if we did have a unit test for it, or a separate E2E test for it, it wouldn't give us the confidence that it would work in each page it's supposed to. So, in the regard, I think even though it's repetitive, it does make sense to do. I guess it depends how important this functionality is and how well we want it tested.